src/main/resources/kweb/kweb_bootstrap.js

```diff
@@ -203,6 +203,62 @@ function removeClass(el, className) {
     }
 }
 
+class DiffPatchData {
+    constructor(prefixEndIndex, postfixOffset, diffString) {
+        this.prefixEndIndex = prefixEndIndex;
+        this.postfixOffset = postfixOffset;
+        this.diffString = diffString;
+    }
+}
+
+function get_diff_changes(id) {
+    let e = document.getElementById(id);
+    let newString = e.dataset.value;//reads the new string value from data-attribute data-vale
+    let oldString = e.dataset.previousInput;//reads the oldString from the data-attribute data-previous-input
+
+    //This seems like a logical place to do this, but I'm unsure if it is safe to do so
+    //I think it might prematurely change the value of toBind.value in prelude.kt's setValue function
+    //savePreviousInput(newString, id)//put the newString into the data attribute so it can be used as the oldString the next time this method is run
+
+    if (oldString === undefined) {//the first time this is run previous-input should be undefined so we just return the new string
+        return new DiffPatchData(0, 0, newString);
+    }
+    let oldArray = Array.from(oldString);
```

Comment:
Why the conversion to Arrays - can the strings not be iterated over directly, eg. using `str.charAt()`?

Reply:
I just prefer working with arrays. I made a mental note to convert it to using str.charAt() after I had the logic fleshed out, and kind of forgot to do that. It should be a quick fix for me to go and change it though.